lldb/source/Plugins/OperatingSystem/Swift/OperatingSystemSwift.cpp

```diff
@@ -0,0 +1,164 @@
+//===-- OperatingSystemSwift.cpp -----------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#if LLDB_ENABLE_SWIFT
+
+#include "OperatingSystemSwift.h"
+
+#include "Plugins/Process/Utility/ThreadMemory.h"
+#include "lldb/Core/Debugger.h"
+#include "lldb/Core/Module.h"
+#include "lldb/Core/PluginManager.h"
+#include "lldb/Target/Process.h"
+#include "lldb/Target/Thread.h"
+#include "lldb/Target/ThreadList.h"
+#include "lldb/Utility/LLDBLog.h"
+#include "lldb/Utility/StructuredData.h"
+
+#include "swift/Threading/ThreadLocalStorage.h"
+
+#include <memory>
+
+using namespace lldb;
+using namespace lldb_private;
+
+LLDB_PLUGIN_DEFINE(OperatingSystemSwift)
+
+void OperatingSystemSwift::Initialize() {
+  PluginManager::RegisterPlugin(GetPluginNameStatic(),
+                                GetPluginDescriptionStatic(), CreateInstance,
+                                nullptr);
+}
+
+void OperatingSystemSwift::Terminate() {
+  PluginManager::UnregisterPlugin(CreateInstance);
+}
+
+OperatingSystem *OperatingSystemSwift::CreateInstance(Process *process,
+                                                      bool force) {
+  if (!process)
+    return nullptr;
+
+  /// Heuristic to only load the plugin for swift programs.
+  FileSpec swift_module_name = FileSpec("libswiftCore.dylib");
+  if (process->GetTarget().GetImages().FindFirstModule(swift_module_name))
+    return new OperatingSystemSwift(*process);
+  return nullptr;
+}
+
+llvm::StringRef OperatingSystemSwift::GetPluginDescriptionStatic() {
+  return "Operating system plug-in converting Swift Tasks into Threads.";
+}
+
+OperatingSystemSwift::~OperatingSystemSwift() = default;
+
+OperatingSystemSwift::OperatingSystemSwift(lldb_private::Process &process)
+    : OperatingSystem(&process) {
+  size_t ptr_size = process.GetAddressByteSize();
+  // Offset of the Task pointer in a Thread's local storage.
+  m_task_ptr_offset_in_tls =
+      swift::tls_get_key(swift::tls_key::concurrency_task) * ptr_size;
+  // Offset of a Task ID inside a Task data structure, guaranteed by the ABI.
+  // See Job in swift/RemoteInspection/RuntimeInternals.h.
+  m_task_id_offset = 4 * ptr_size + 4;
```

Comment:
You should check that the swift concurrency version is still 1 here.

See: https://github.com/swiftlang/swift/pull/76437